packages/mobile/src/walletConnect/saga.ts

```diff
@@ -0,0 +1,283 @@
+import { EncodedTransaction } from '@celo/connect'
+import { GenesisBlockUtils } from '@celo/network-utils'
+import { UnlockableWallet } from '@celo/wallet-base'
+import AsyncStorage from '@react-native-community/async-storage'
+import '@react-native-firebase/database'
+import '@react-native-firebase/messaging'
+import WalletConnectClient, { CLIENT_EVENTS } from '@walletconnect/client'
+import { PairingTypes, SessionTypes } from '@walletconnect/types'
+import { EventChannel, eventChannel } from 'redux-saga'
+import { spawn } from 'redux-saga-test-plan/matchers'
+import { call, put, select, take, takeEvery, takeLeading } from 'redux-saga/effects'
+import { readGenesisBlockFile } from 'src/geth/geth'
+import networkConfig from 'src/geth/networkConfig'
+import { navigate, navigateBack } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import Logger from 'src/utils/Logger'
+import {
+  AcceptRequest,
+  AcceptSession,
+  Actions,
+  clientInitialised,
+  CloseSession,
+  closeSession as closeSessionAction,
+  DenyRequest,
+  DenySession,
+  initialiseClient as initialiseClientAction,
+  InitialisePairing,
+  initialisePairing as initialisePairingAction,
+  pairingCreated,
+  pairingDeleted,
+  pairingProposal,
+  pairingUpdated,
+  requestFulfilled,
+  sessionCreated,
+  sessionDeleted,
+  SessionPayload,
+  sessionPayload,
+  SessionProposal,
+  sessionProposal,
+  sessionUpdated,
+} from 'src/walletConnect/actions'
+import { SupportedActions } from 'src/walletConnect/constants'
+import { getWallet } from 'src/web3/contracts'
+import { getAccountAddress, unlockAccount } from 'src/web3/saga'
+import { currentAccountSelector } from 'src/web3/selectors'
+
+const TAG = 'WalletConnect/saga'
+
+let client: WalletConnectClient | null = null
+
+export function* acceptSession({ proposal }: AcceptSession) {
+  if (!client) {
+    Logger.debug(TAG + '@acceptSession', 'missing client')
+    return
+  }
+
+  const { nodeDir } = networkConfig
+  const genesis: string = yield call(readGenesisBlockFile, nodeDir)
+  const networkId: number = GenesisBlockUtils.getChainIdFromGenesis(genesis)
```

Comment:
This is nice because it's robust but wondering if it wouldn't be simpler and faster to just have a chainid in each network's env file?
https://github.com/celo-org/wallet/blob/main/packages/mobile/.env.mainnet

Reply:
Yeah definitely. Didn't think to do it like that